### ISSUE TITLE
TRS Data

### ISSUE THREAD

**Issue by Jegelewicz** [ArctosDB/arctos#2076]
> I have a question on TRS input into Arctos. On the Locality handbook page, it states under the Specific Locality Section to not put in TRS, but then lower down in the TRS is states to put the TRS in specific locality and then convert. What should be down with TRS data? Most of our localities have them.
>  
> Thanks

Lindsey Frederick

Bioscience Collections Manager
New Mexico Museum of Natural History and Science

**Comment by Jegelewicz** [comment]
At UTEP, I put the TRS data in Locality remarks unless it was the only "specific locality" I had. I have since learned that this information should at the very least be included in verbatim locality if you want it to remain associated with the specimen.

Anyone else have input?

**Comment by dustymc** [comment]
I would put them in verbatim locality or locality remarks if it's not something that could be considered verbatim.

It is useful to have specific locality as "machine readable" as possible; TRS data there will definitely confuse eg automatic georeferencing services.

I'll update the locality documentation - it's referring to a very old model.

**Comment by lin-fred** [comment]
Is this the same issue still open on TRS documentation ArctosDB/arctos#995?

And the handbook might need to be updated if this is the official recommendation at the moment
https://handbook.arctosdb.org/documentation/locality.html#trs

Also, the link to convert TRS doesn't seem to work anymore? Any other recommendations on conversion?

https://handbook.arctosdb.org/documentation/locality.html#locality
The bullet-point right before Locality nickname, that has a link for Verbatim Locality, sends me to a 404.

**Comment by dustymc** [comment]
>#995

Thanks/closed.

>link to convert TRS doesn't seem to work 

I noticed that, but I'm not sure what to do - maybe it's temporary??

>Verbatim Locality

Thanks/fixed.

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
Lindsey just brought this issue up to me. We also have a lot of TRS data in paleo (at least 8,000 of our localities have them), and I think it would be useful to retain the format of these coordinates rather than just lumping them together into a string of text.  In my experience a lot of paleo folk still use this grid system, even if they are not using it as a way to accurately mark coordinates. Would it be possible to add a TRS format option to original units? Basically, I want to be able to parse out and sort township, range, and section in a search.

I may have to open another issue for this, but can more than one original unit be added?

> link to convert TRS doesn't seem to work

What I've used GeoLocate in the past to convert TRS to lat/long. Would it be useful for me to write up a how-to for this?

**Comment by dustymc** [comment]
>TRS format option to original units

Yes, sorta - but those data are stored as a semi-formatted string.

>parse out and sort township, range, and section in a search.

Maybe we need to store "verbatim" as JSON or something? That would allow structure-on-demand, a place to specify where the data came from, etc. I think Oracle12 would be required.

>how-to

Yes please.

There are a bunch of similar issues that should be consolidated and discussed. 

https://github.com/ArctosDB/arctos/issues/1913
https://github.com/ArctosDB/arctos/issues/752
https://github.com/ArctosDB/arctos/issues/2041
https://github.com/ArctosDB/arctos/issues/1971
https://github.com/ArctosDB/arctos/issues/969

**Comment by Jegelewicz** [comment]
> Maybe we need to store "verbatim" as JSON or something?

I may be completely dumb here, but everyone with an Excel spreadsheet or Access database can create the field TRS or three fields Township, Range, Section. It is hard to understand why we can't just add those fields to collecting event, even if they don't DO anything EXCEPT let people find everything from any given TRS or individual Township, etc. These are just like UTM, which we have added as fields in collecting event, correct? If TRS and UTM are non-mapping "coordinate" fields in the collecting event, then we can record them as they are without mashing them up in a big string of text.

> can more than one original unit be added?

UTM and TRS can still be useful to those who use them and they don't mess with "mappable" coordinates. I don't see why we can't have them along with "mappable" coordinates.

**Comment by Jegelewicz** [comment]
Next best idea - make them event attributes.

**Comment by campmlc** [comment]
I prefer this idea. Do we have so e limit in available fields? If not, this
makes the most sense.

**Comment by dustymc** [comment]
>can't just add those fields to collecting event

Normalization and usability mostly. 

>event attributes

Works for me. Some sort of big-picture roadmap would still be exceedingly useful.

>have so e limit in available fields?

Not one you're likely to notice, but that's not a usually design consideration.

If you meant for 'event attributes' they will be normalized, not Excel-nightmare-format!

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
> > how-to

> Yes please.

I've created a document on how to covert TRS to lat/long using geolocate. How/where should I upload this?

**Comment by Jegelewicz** [comment]
@Nicole-Ridgwell-NMMNHS is it a Word Doc? You can send it to me and I can add it to the documentation.

Thanks!!!!

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
> @Nicole-Ridgwell-NMMNHS is it a Word Doc? You can send it to me and I can add it to the documentation.
> 
> Thanks!!!!

Done. We should replace the non-functioning link in the locality handbook page with a link to the how-to.

I'm fine with making TRS an event attribute.

**Comment by Jegelewicz** [comment]
@Nicole-Ridgwell-NMMNHS check out [your documentation](https://handbook.arctosdb.org/how_to/Convert_TRS_to_Lat-Long_Using_GeoLocate.html) and let me know if anything appears crazy.

Thank you so much for writing this!!!

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
Looks great! Thanks for adding the screenshots!

**Comment by Jegelewicz** [comment]
OK, we have some decent documentation. Now, I propose that we add non-mapping, verbatim TRS data to the collecting event, just as we have done with UTM.

**Comment by dustymc** [comment]
> just like UTM, which we have added as fields in collecting event, correct? 

Not really; there's some temp stuff hanging around, but "verbatim_coordinates" is "the data field." 

>limit in available fields

Event Attributes will work the same as part and specimen attributes - there's a limit (ca. 30! "fields" in the current version), but it's nothing we'll ever encounter.

I think event attributes is a good place for this, and any other coordinate-like verbatim-ish data. It should let individual users or collections store whatever they want in whatever format they want without denormalizing the core data, scattering "verbatim" data around so that it can't be found by users, or confounding "as entered" and "as provided by the collector."

**Comment by Jegelewicz** [comment]
> I think event attributes is a good place for this, and any other coordinate-like verbatim-ish data. It should let individual users or collections store whatever they want in whatever format they want without denormalizing the core data, scattering "verbatim" data around so that it can't be found by users, or confounding "as entered" and "as provided by the collector."

I like it. I think we need to compile a list of the proposed event attributes and get this underway.

ArctosDB/arctos#1270 
ArctosDB/arctos#1058

**Comment by dustymc** [comment]
>compile a list of the proposed event attributes

Meh, it's just data - we can add them whenever we want.

>get this underway

It is. Most of the background stuff exists but hasn't been tested, I'm working on UI now. It's a little slow since I have to play nice with production data, but I should have something soon-ish.
